utilities/transactions/write_committed_transaction_ts_test.cc

```diff
@@ -538,6 +538,81 @@ TEST_P(WriteCommittedTxnWithTsTest, GetForUpdate) {
   txn5.reset();
 }
 
+TEST_P(WriteCommittedTxnWithTsTest, GetForUpdateUdtValidationNotEnabled) {
+  ASSERT_OK(ReOpenNoDelete());
+
+  ColumnFamilyOptions cf_options;
+  cf_options.comparator = test::BytewiseComparatorWithU64TsWrapper();
+  const std::string test_cf_name = "test_cf";
+  ColumnFamilyHandle* cfh = nullptr;
+  assert(db);
+  ASSERT_OK(db->CreateColumnFamily(cf_options, test_cf_name, &cfh));
+  delete cfh;
+  cfh = nullptr;
+
+  std::vector<ColumnFamilyDescriptor> cf_descs;
+  cf_descs.emplace_back(kDefaultColumnFamilyName, options);
+  cf_descs.emplace_back(test_cf_name, Options(DBOptions(), cf_options));
+  options.avoid_flush_during_shutdown = true;
+
+  txn_db_options.enable_udt_validation = false;
+  ASSERT_OK(ReOpenNoDelete(cf_descs, &handles_));
+
+  // blind write a key/value for latter read via `GetForUpdate`.
+  std::unique_ptr<Transaction> txn0(
+      NewTxn(WriteOptions(), TransactionOptions()));
+  ASSERT_OK(txn0->Put(handles_[1], "key", "value0"));
+  ASSERT_OK(txn0->SetCommitTimestamp(20));
+  ASSERT_OK(txn0->Commit());
+
+  // When timestamp validation is disabled across the whole DB
+  // `SetReadTimestampForValidation` should not be called.
+  std::unique_ptr<Transaction> txn1(
+      NewTxn(WriteOptions(), TransactionOptions()));
+  std::string value;
+  ASSERT_OK(txn1->SetReadTimestampForValidation(21));
+  ASSERT_TRUE(txn1->GetForUpdate(ReadOptions(), handles_[1], "key", &value,
+                                 /* exclusive= */ true, /*do_validate=*/true)
+                  .IsInvalidArgument());
+  txn1.reset();
+
+  // do_validate and no snapshot, no conflict checking at all
+  std::unique_ptr<Transaction> txn2(
+      NewTxn(WriteOptions(), TransactionOptions()));
+  ASSERT_OK(txn2->GetForUpdate(ReadOptions(), handles_[1], "key", &value,
+                               /* exclusive= */ true, /*do_validate=*/true));
+  ASSERT_OK(txn2->Put(handles_[1], "key", "value1"));
+  ASSERT_OK(txn2->SetCommitTimestamp(21));
+  ASSERT_OK(txn2->Commit());
+  txn2.reset();
+
+  // do_validate and set snapshot, execute sequence number based conflict
+  // checking and skip timestamp based conflict checking.
```

Comment:
Would it make sense to test (using parallel txns) that timestamp based conflicts are *not* caught if the new option is false?

Reply:
That's a good idea! I added a test to write some out of order timestamps.